src/home_robot_sim/home_robot_sim/env/habitat_goat_env/habitat_goat_env.py

```diff
@@ -0,0 +1,212 @@
+
+        print("Initializing perception module with vocabulary:", vocabulary)
+
+    def _preprocess_obs(
+        self, habitat_obs: habitat.core.simulator.Observations
+    ) -> home_robot.core.interfaces.Observations:
+        depth = self._preprocess_depth(habitat_obs["depth"])
+        goals, vocabulary = self._preprocess_goals(
+            self.current_episode.tasks, habitat_obs
+        )
+        obs = home_robot.core.interfaces.Observations(
+            rgb=habitat_obs["rgb"],
+            depth=depth,
+            compass=habitat_obs["compass"],
+            gps=self._preprocess_xy(habitat_obs["gps"]),
+            task_observations={
+                "tasks": goals,
+                "top_down_map": self.get_episode_metrics()["goat_top_down_map"],
+            },
+            camera_pose=None,
+            third_person_image=None,
+        )
+        obs = self._preprocess_semantic(obs, habitat_obs["semantic"], vocabulary)
+        return obs
+
+    def _preprocess_semantic(
+        self,
+        obs: home_robot.core.interfaces.Observations,
+        habitat_semantic: np.ndarray,
+        vocabulary,
+    ) -> home_robot.core.interfaces.Observations:
+        if self.ground_truth_semantics:
+            instance_id_to_category_id = (
+                self.semantic_category_mapping.instance_id_to_category_id
+            )
+            obs.semantic = instance_id_to_category_id[habitat_semantic[:, :, -1]]
+            obs.task_observations["instance_map"] = habitat_semantic[:, :, -1] + 1
+
+            for idx_cat, obj_cat in enumerate(vocabulary):
+                obj_cat = " ".join(obj_cat.split("_"))
+                try:
+                    if obj_cat in self.semantic_category_mapping.all_hm3d_categories:
+                        idx = self.semantic_category_mapping.all_hm3d_categories.index(
+                            obj_cat
+                        )
+                        obs.semantic[obs.semantic == idx] = -1 * (idx_cat + 1)
+                except Exception as e:
+                    print(e)
+                    import pdb
+
+                    pdb.set_trace()
+
+            obs.semantic[obs.semantic >= 0] = 0
+            obs.semantic = obs.semantic * -1
+            # TODO Ground-truth semantic visualization
+            obs.task_observations["semantic_frame"] = obs.rgb
+        else:
+            obs = self.segmentation.predict(obs, depth_threshold=0.5)
+
+        obs.task_observations["semantic_frame"] = np.concatenate(
+            [obs.rgb, obs.semantic[:, :, np.newaxis]], axis=2
+        ).astype(np.uint8)
+        return obs
+
+    def _preprocess_depth(self, depth: np.array) -> np.array:
+        rescaled_depth = self.min_depth + depth * (self.max_depth - self.min_depth)
+        rescaled_depth[depth == 0.0] = MIN_DEPTH_REPLACEMENT_VALUE
+        rescaled_depth[depth == 1.0] = MAX_DEPTH_REPLACEMENT_VALUE
+        return rescaled_depth[:, :, -1]
+
+    def _preprocess_goals(self, tasks, habitat_obs):
```

Comment:
We need to design a clean API between agent and environment. 

The environment should only specify the task:
- at initialization: specify object instances by images ("my desk", "my mug", "Mukul")
- in `get_observation()`, specify goals only as long-form English instructions (like "Can you find my keys? I think I forgot them on my desk in my bedroom. Otherwise, they're probably under a pillow on the sofa." or "Where is Sara? I think she's reading in the living room. Otherwise, she's probably in the bed. Can you find and follow her?")

The agent should contain all the logic:
- LLM parsing of long-form English instruction into low-level skills 
- mapping
- skill execution
- etc.